### PR TITLE
feat: Instance commands

### PR DESCRIPTION
**Pull request by fgerschwiler**

### CODE REVIEW

File: internal/command/instance_converter.go [hunk 1]
```diff
@@ -205,3 +205,10 @@ func writeModelToFeatures(wm *FeaturesWriteModel) *domain.Features {
 		MaxActions:               wm.MaxActions,
 	}
 }
+
+func instanceDomainWriteModelToInstanceDomain(wm *InstanceDomainWriteModel) *domain.InstanceDomain {
+	return &domain.InstanceDomain{
+		ObjectRoot: writeModelToObjectRoot(wm.WriteModel),
+		Domain:     wm.Domain,
+	}
+}
```
Comment (livio-a) [inline internal/command/instance_converter.go:214]: never used

File: internal/command/instance_model.go [hunk 2]
```diff
@@ -1,15 +1,20 @@
...
 	"github.com/caos/zitadel/internal/repository/instance"
-	"golang.org/x/text/language"
 )
 
 type InstanceWriteModel struct {
 	eventstore.WriteModel
 
+	Name            string
+	State           domain.InstanceState
+	GeneratedDomain string
```
Comment (livio-a) [inline internal/command/instance_model.go:16]: just a little thing, but do we really need the generated domain on the write model? what do we ensure with that
same for existing projectID and globalOrgID

File: internal/domain/instance.go [hunk 3]
```diff
@@ -11,6 +11,9 @@ const (
 type Instance struct {
 	models.ObjectRoot
 
+	Name                            string
```
Comment (livio-a) [inline internal/domain/instance.go:14]: we never use the Instance model, i'd only specify the state here

File: internal/domain/instance_domain.go [hunk 4]
```diff
@@ -0,0 +1,36 @@
+package domain
+
+import (
+	"strings"
+
+	"github.com/caos/zitadel/internal/eventstore/v1/models"
+)
+
+type InstanceDomain struct {
```
Comment (livio-a) [inline internal/domain/instance_domain.go:9]: we never use the domain model, i'd only specify the state here

File: internal/repository/instance/instance.go [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+
+	Name string `json:"name,omitempty"`
+}
+
+func (e *InstanceAddedEvent) Data() interface{} {
+	return e
+}
+
+func (e *InstanceAddedEvent) UniqueConstraints() []*eventstore.EventUniqueConstraint {
+	return nil
```
Comment (livio-a) [inline internal/repository/instance/instance.go:29]: do we need a unique constraint on the name?

**Review by livio-a — COMMENTED** [review]
passing the instanceID to the various instance commands / writemodels makes sense

but some commands for org / users / ... now also have a separate instanceID parameter for getting some default policy
IMO on org/user/project/... level we should either read the instanceID from context or pass it to every command and the corresponding writemodel. Otherwise it's not consistent and only confusing

File: internal/command/instance_model.go (hunk 2 above)
Comment (fgerschwiler) [reply internal/command/instance_model.go:16]: Not sure, we will have a look at this with @adlerhurst branch

File: internal/domain/instance.go (hunk 3 above)
Comment (fgerschwiler) [reply internal/domain/instance.go:14]: I will remove this, if we need it this will come with the branch of @adlerhurst

**Review by livio-a — APPROVED** [review]

**Pull request merged**